### PR TITLE
feat: add Not Found page bb-51

### PR DESCRIPTION
**Pull request by yulyaolshanska**
### Summary
This PR implements the Not Found page.

### Created
- Not Found page component
- getValidClassNames helper

### Changes
- **link.tsx:** Added type prop for Link component
- Reset margin in global styles

image

#### Hover 
image

### CODE REVIEW

File: apps/frontend/src/index.tsx [hunk 1]
```diff
@@ -38,6 +39,7 @@ createRoot(document.querySelector("#root") as HTMLElement).render(
 							},
 						],
 						element: <App />,
+						errorElement: <NotFound />,
```
Comment (fshabanov) [inline apps/frontend/src/index.tsx:42]: Instead add a new route for path "*"

File: apps/frontend/src/libs/components/link/link.tsx [hunk 2]
```diff
@@ -5,11 +5,18 @@ import { type ValueOf } from "~/libs/types/types.js";
 
 type Properties = {
 	children: React.ReactNode;
+	className?: string;
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/link/link.tsx:8]:
```suggestion
```
Make it customizable using props instead

File: apps/frontend/src/pages/not-found/not-found.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+import styles from "./styles.module.css";
+
+const NotFound: React.FC = () => (
+	<div className={styles["pageContainer"]}>
+		<h1 className={styles["title"]}>404 - Page Not Found</h1>
+		<p className={styles["text"]}>
+			The page you are looking for does not exist.
+		</p>
+		<Link className={styles["button"] || ""} to={AppRoute.ROOT}>
```
Comment (fshabanov) [inline apps/frontend/src/pages/not-found/not-found.tsx:12]:
```suggestion
		<Link className={styles["button"]} to={AppRoute.ROOT}>
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/index.tsx (hunk 1 above)
Comment (yulyaolshanska) [reply apps/frontend/src/index.tsx:42]: image

I added Not Found like this. Is it ok that i see navigation?
``` {
								element: <NotFound />,
								path: "*",
							},
```

File: apps/frontend/src/libs/components/link/link.tsx (hunk 2 above)
Comment (yulyaolshanska) [reply apps/frontend/src/libs/components/link/link.tsx:8]: I don't quite understand what that means. I need to add for example prop type and use like this?
```
type Properties = {
	type: string;
}

	<NavLink className={clsx(styles["link"], type)} to={to}>

```

**Review by WilhelmWesser — APPROVED** [review]

**Review by revenets — APPROVED** [review]

File: apps/frontend/src/index.tsx (hunk 1 above)
Comment (fshabanov) [reply apps/frontend/src/index.tsx:42]: No, you shouldn't see the navigation

File: apps/frontend/src/libs/components/link/link.tsx (hunk 2 above)
Comment (fshabanov) [reply apps/frontend/src/libs/components/link/link.tsx:8]: Yes, this approach is better. But be more specific with the type of `type` variable, not just string

File: apps/frontend/src/index.tsx [hunk 4]
```diff
@@ -40,6 +41,10 @@ createRoot(document.querySelector("#root") as HTMLElement).render(
 						element: <App />,
 						path: AppRoute.ROOT,
 					},
+					{
+						element: <NotFound />,
+						path: "*",
```
Comment (fshabanov) [inline apps/frontend/src/index.tsx:46]:
```suggestion
						path: AppRoute.ANY,
```

File: apps/frontend/src/libs/components/link/link.tsx [hunk 5]
```diff
@@ -1,15 +1,21 @@
+import clsx from "clsx";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/link/link.tsx:1]: Create a helper `getValidClassNames`

File: apps/frontend/src/libs/types/components/link-type.type.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+type LinkType = "button" | "navLink";
```
Comment (fshabanov) [inline apps/frontend/src/libs/types/components/link-type.type.ts:1]: Looks like a type for link component only. Move to ~/libs/components/link/libs/types.ts

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/link/libs/types.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+type LinkType = "button" | "navLink";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/link/libs/types.ts:1]: Create types folder as well, to have same structure as in the rest of app

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Pull request merged**